File: app/src/main/java/com/example/quicknotes/model/Tag.java
package com.example.quicknotes.model;
import androidx.annotation.ColorRes;

public class Tag {
    private final String name;
    @ColorRes private final int colorResId;

    public Tag(String name, @ColorRes int colorResId) {
        this.name = name;
        this.colorResId = colorResId;
    }

    public String getName() {
        return name;
    }
    public int getColorResId() {
        return colorResId;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Tag)) return false;
        Tag other = (Tag) o;
        return name.equalsIgnoreCase(other.name);
    }

    @Override
    public int hashCode() {
        return name.toLowerCase().hashCode();
    }

}

File: app/src/main/java/com/example/quicknotes/model/Tag.kt
package com.example.quicknotes.model;
import androidx.annotation.ColorRes;

public record Tag(String name, @ColorRes int colorResId) {
    @Override
    public int colorResId() {
        return colorResId;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Tag other)) return false;
        return name.equalsIgnoreCase(other.name);
    }

    @Override
    public int hashCode() {
        return name.toLowerCase().hashCode();
    }

}

